docs/rules/no-promise-in-fire-event.md

```diff
@@ -0,0 +1,31 @@
+# Disallow the use of promises passed to a `fireEvent` method (no-promise-in-fire-event)
+
+The `fireEvent` method expects that a DOM element is passed.
+
+Examples of **incorrect** code for this rule:
+
+```js
+import { screen, fireEvent } from '@testing-library/react';
+
+// usage of findBy queries
+fireEvent.click(screen.findByRole('button'));
+
+// usage of promises
+fireEvent.click(new Promise(jest.fn())
```

Comment:
Does this cover the scenario where the promise is in a variable ? Like

```ts
const promise = new Promise(jest.fn())
fireEvent.click(promise)
```

If so, perhaps we should add it here in the docs

Reply:
That's great feedback!
It doesn't cover this case, not sure how we can check that :/